components/data_comps/dice/bld/build-namelist

```diff
@@ -306,6 +306,9 @@ if (defined $opts{'infile'}) {
 
 my %xmlvars = ();
 SetupTools::getxmlvars(${CASEROOT},\%xmlvars);
+# need to expand DIN_LOC_ROOT first
+$xmlvars{DIN_LOC_ROOT}=SetupTools::expand_xml_var($xmlvars{DIN_LOC_ROOT}, \%xmlvars);
```

Comment:
If this variable isn't resolved first the result in %xmlvars is corrupted.   This is the bandaid.